[OCPDOCS#19136]: CQA 2.0 configuring arbiter node (#115237)
openshift/openshift-docs: modules/installing-ocp-agent-local-arbiter-node.adoc

// Module included in the following assemblies:
//
// * installing/installing_with_agent_based_installer/preparing-to-install-with-agent-based-installer.adoc

:_mod-docs-content-type: CONCEPT
[id="installing-ocp-agent-local-arbiter-node_{context}"]

> = About local arbiter node

Reviewer: This is a solid title. Preference-wise, maybe keeping the word configure somewhere in here would be better, but I am not sure.:

```suggestion
=  Local arbiter node configuration
```

Up to you

Reviewer: If you do keep it, I think without an article like "a" or "the" it would make sense to pluralize node:

```suggestion
= About local arbiter nodes
```

Author: Yeah. I thought about it. I'm more inclined towards an "About a local..." title for the concept. Having "configure" in there makes sense, but as we have removed the procedure steps, I believe it should not give the idea that it is a process.

* You can configure a local arbiter node only. Remote arbiter nodes are not supported.
* The arbiter node must be located in a network environment with an end-to-end latency of less than 500 milliseconds, including disk I/O. In high-latency environments, you might need to apply the `etcd` slow profile.
* The control plane nodes must also have enough storage for the workload.
====

To deploy a cluster with two control plane nodes and one local arbiter node, you must define the following nodes in the `install-config.yaml` file:
For the minimum system requirements of the control plane nodes and arbiter node, see "Recommended resources for topologies" section.

* 2 control plane nodes
* 1 arbiter node

The arbiter node must meet the following minimum system requirements:

* 2 vCPUs
* 8 GB of RAM
* 50 GB of SSD or equivalent storage

The arbiter node must be located in a network environment with an end-to-end latency of less than 500 milliseconds, including disk I/O. In high-latency environments, you might need to apply the `etcd` slow profile.

The control plane nodes must meet the following minimum system requirements:

* 4 vCPUs
* 16 GB of RAM
* 120 GB of SSD or equivalent storage

> Additionally, the control plane nodes must also have enough storage for the workload.

Reviewer: I think this content before the prereqs section is fine to keep, even if it's a bit redundant with other content on this page. I would revert this change just so we're not messing around too much with what is technically someone else's doc, I only want to convert this to a concept with as few changes as possible.

(also when reverting this, remove the two extra list items in the admonition and the line referencing the "Recommended resources for topologies" section)

Reviewer: Also to be clear, I don't think these are bad changes to make at all, these in particular are pretty logical considering where you placed this content. I just want to prioritize moving the content rather than having to touch it up too much

> .Example `install-config.yaml` configuration for deploying an arbiter node

Author: Not sure if I should have removed this example and callouts, as this is a concept topic, but I kept it anyway for the purpose of how the configuration would work. WDYT?

Reviewer: This example is pretty crucial to keep, as I feel like it's the main reason to document this particular content. So it's good that you kept it in place, and that you removed callouts while you were at it.

        role: arbiter
# ...
----
where:

`arbiter`:: Specifies the arbiter machine pool. You configure this field to deploy a cluster with an arbiter node.

> `arbiter.replicas`:: Specifies the number of machines in the arbiter pool. Set this field to `1`.

Reviewer: This is a good rewrite. In the spirit of doing minimal changes though, I would try to keep as much of the original text for now:

```suggestion
`arbiter.replicas`:: Specifies the `arbiter.replicas` parameter as `1` for the arbiter pool. You cannot set this field to a value that is greater than 1.
```

`arbiter.name`:: Specifies a name for the arbiter machine pool.
`controlPlane`:: Specifies the control plane machine pool.

> `controlPlane.replicas`:: Specifies the control plane machines. When an arbiter pool is defined, two control plane replicas are valid.

Reviewer: Also a good rewrite, but also would recommend keeping the changes minimal and as non-meaning-changing as possible:

```suggestion
`controlPlane.replicas`:: Specifies the `controlPlane.replicas` parameter. When an arbiter pool is defined, two control plane replicas are valid.
```
